src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -643,13 +643,16 @@ private static void PartialMethodChecks(SourceOrdinaryMethodSymbol definition, S
             Debug.Assert(!ReferenceEquals(definition, implementation));
 
             MethodSymbol constructedDefinition = definition.ConstructIfGeneric(TypeMap.TypeParametersAsTypeSymbolsWithIgnoredAnnotations(implementation.TypeParameters));
-            bool returnTypesEqual = constructedDefinition.ReturnTypeWithAnnotations.Equals(implementation.ReturnTypeWithAnnotations, TypeCompareKind.AllIgnoreOptions);
-            if (!returnTypesEqual
-                && !SourceMemberContainerTypeSymbol.IsOrContainsErrorType(implementation.ReturnType)
-                && !SourceMemberContainerTypeSymbol.IsOrContainsErrorType(definition.ReturnType))
+            bool hasTypeDifferences = !constructedDefinition.ReturnTypeWithAnnotations.Equals(implementation.ReturnTypeWithAnnotations, TypeCompareKind.AllIgnoreOptions);
```

Comment:
Does this introduce new errors when one of the partial declarations has an error type for its return type?

Reply:
> Does this introduce new errors when one of the partial declarations has an error type for its return type?

Yes, I believe so, but if the return types are distinct, arguably we should report a warning, even if one or both of those return types is an error type.